### PR TITLE
fix: serialize dispatch slot admission

### PR DESCRIPTION
**Pull request by therockstorm**
## Why

Two concurrent `crew start` processes could each observe an open slot before either persisted its run. With `maximumInProgress: 1`, both could launch, exceeding the configured orchestration limit and consuming more agent capacity than operators allowed.

## Summary

- serialize cross-process slot admission under one state-root lock
- re-read active provisioning/running runs while holding that lock and persist the provisioning reservation before releasing it
- add a deterministic two-process regression test for distinct task slugs at a maximum of one

## Validation

- `npx vitest run e2e/dispatch.e2e.test.ts -t "reserves one shared slot"`
- `npx vitest run e2e/dispatch.e2e.test.ts` (46 passed)
- `node --run verify` (79 passed, 1 skipped)

## Notes

- Validates and resolves both duplicate findings: `fnd_sig-feat-cli-command-49a9766219-_11fd05c55d` and `fnd_sig-feat-cli-command-012418b101-_4055fbdd6c`.
- Review focus: admission lock scope and durable reservation semantics.

Agent session: `codex resume 019fecf3-dfb5-7931-ae12-80611ee76626`

<sub>🤖 <code>cb-ship:created v1 skill@1.0.2</code></sub>

### CODE REVIEW

File: v2/src/run/index.ts [hunk 1]
```diff
@@ -598,6 +624,29 @@ class RunStore {
...
+        ).length;
+        if (!input.force && activeCount >= input.maximumInProgress) {
+          return { activeCount };
+        }
+        const record = await this.create(input);
+        return { activeCount, record };
+      },
+      path: join(this.#runsDirectory, "dispatch-admission.lock"),
+    });
+  }
```
[the remark this answers is not in the document]
Comment (therockstorm) [reply v2/src/run/index.ts:648]: Addressed in https://github.com/ClipboardHealth/groundcrew/commit/fea5b8e38b81d699175c0110716c8bfa2beaaaf8. Moved dispatch admission locks into a dedicated runs/.locks namespace and added a regression test for dispatch:admission.

<sub>🤖 <code>cb-babysit:addressed v1 skill@1.0.4</code></sub>

**Comment by therockstorm** [comment]
## Review-body findings

**Agree**

- CodeRabbit: The admission lock could collide with the valid `dispatch:admission` task slug. Fixed in https://github.com/ClipboardHealth/groundcrew/commit/fea5b8e38b81d699175c0110716c8bfa2beaaaf8 by moving admission locks into a dedicated namespace and adding a regression test.

**Already fixed**

- Mendral reported LGTM with no actionable findings; no additional change was requested.

```text
b0f10785e88d1641
fd6ab370b6a3824c
```

<sub>🤖 <code>cb-babysit:addressed v1 skill@1.0.4</code></sub>
